tests/resources/endpoints/cadip_search.yaml

```diff
@@ -0,0 +1,164 @@
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+stations:
+  cadip:
+    endpoint:
+      url: http://127.0.0.1:8000
+      secret:
+        username: test
+        password: test
+collections:
+  - id: cadip_session_by_id_list
+    station: cadip
+    query:
+      id: S1A_20170501121534062343,S1A_20240328185208053186
+      limit: 20
+  - id: cadip_session_by_id
+    station: cadip
+    query:
+      id: S1A_20240328185208053186
+      limit: 20
+
+  - id: cadip_session_by_id_platform
+    station: cadip
+    query:
+      id: S1A_20240328185208053186
+      satellite: S1A
+      limit: 20
+
+  - id: cadip_session_by_lists_id_platform
+    station: cadip
+    query:
+      id: S1A_20240328185208053186,S1A_20240328185208053186
+      satellite: S1A,S2B
+      limit: 20
+
+  - id: cadip_session_by_platform_list
+    station: cadip
+    query:
+      satellite: S1A, S2B
+      limit: 20
+
+  - id: cadip_session_by_start_stop_platform
+    station: cadip
+    query:
+      start_date: "2020-02-16T12:00:00Z"
+      stop_date: "2023-02-16T12:00:00Z"
+      satellite: S1A
+      limit: 20
+
+  - id: cadip_session_by_interval
+    station: cadip
+    query:
+      start_date: "2014-02-16T12:00:00Z"
+      stop_date: "2024-02-16T12:00:00Z"
+      limit: 2
+
+  - id: cadip_session_by_satellite
+    station: cadip
+    query:
+      satellite: S1A
+      limit: 2
+
+  - id: cadip_session_s2b
+    station: cadip
+    query:
+      satellite: S2B
+
+  - id: cadip_session_incorrect
+    station: cadip
+    query:
+      satellite: incorrect_platform
+
+  - id: cadip_session_incomplete
+    station: cadip
+    query:
+      limit: 1
+
+  - id: cadip_session_incomplete_no_stop
+    station: cadip
+    query:
+      start_date: "2020-02-16T12:00:00Z"
+
+  - id: cadip_session_incomplete_no_start
+    station: cadip
+    query:
+      stop_date: "2020-02-16T12:00:00Z"
+
+  - id: cadip_session_incomplete_platf_no_start
+    station: cadip
+    query:
+      satellite: S1A
+      start_date: "2020-02-16T12:00:00Z"
+
+  - id: cadip_session_incorrect_station
+    station: incorrect
+    query:
+      satellite: S1A
+
+  - id: cadip_valid_auth
+    station: cadip
+    query:
+      limit: 1
+
+  ###
+  - id: cadip
+    station: cadip
+    query:
+      limit: 100
+  - id: ins
+    station: ins
+    query:
+      limit: 100
+  - id: mps
+    station: mps
+    query:
+      limit: 100
+  - id: mti
+    station: mti
+    query:
+      limit: 100
+  - id: nsg
+    station: nsg
+    query:
+      limit: 100
+  - id: sgs
+    station: sgs
+    query:
+      limit: 100
+  - id: s1_ins
+    station: ins
+    query: $filter=Satellite in ('S1A', 'S1C')
```

Comment:
Either we keep my original proposition of including the whole OData $filter syntax in the configuration file, or yours of splitting it key by key, but we can't keep both syntaxes.

Reply:
Will be updated.